abn/core/grpc_service_test.go

```diff
@@ -244,3 +249,51 @@ func getMetric(a *abnapp.Application, track, metric string) *summarymetrics.Summ
 	m, _ := v.GetMetric(metric, true)
 	return m
 }
+
+func ensureSecretCreated(application string) error {
+	namespace := namespaceFromKey(application)
+	name := secretNameFromKey(application)
+	_, err := k8sclient.Client.Typed().CoreV1().Secrets(namespace).Get(context.Background(), name, metav1.GetOptions{})
+	if err == nil {
+		return nil
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	_, err = k8sclient.Client.Typed().CoreV1().Secrets(namespace).Create(context.Background(), secret, metav1.CreateOptions{})
+	return err
```

Comment:
Small nitpick but I think reversing the logic makes more sense in the context of the function

```suggestion
	// if secret has not been created, then create it
	if err != nil {
		secret := &corev1.Secret{
			ObjectMeta: metav1.ObjectMeta{
				Name:      name,
				Namespace: namespace,
			},
		}
		_, err = k8sclient.Client.Typed().CoreV1().Secrets(namespace).Create(context.Background(), secret, metav1.CreateOptions{})
		return err
	}

	return nil
```